python-package/lightgbm/basic.py

```diff
@@ -232,6 +232,7 @@ class _ConfigAliases(object):
                           "metric_types"},
                "num_class": {"num_class",
                              "num_classes"},
+               "start_iteration": {"start_iteration"},
```

Comment:
I don't think we need this line here. Params without aliases are handled in the `get` method.